lib/agents/remote.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const WebDriverAgent = require('../WebdriverAgent');
+const { WebDriver, Capabilities } = require('selenium-webdriver');
+const _http = require('selenium-webdriver/http');
+
+class RemoteAgent extends WebDriverAgent {
+  constructor(options) {
+    super(options);
+
+    if (!options || typeof options.webdriverServer !== 'string') {
+      throw new Error('RemoteAgent: \'webdriverServer\' is required but was ' +
+        'not specified.');
+    }
+    this.webdriverServer = options.webdriverServer;
+
+    if (!options.capabilities) {
+      throw new Error('RemoteAgent: \'capabilities\' is required but was ' +
+        'not specified.');
+    }
+    if (typeof options.capabilities.browserName !== 'string') {
+      throw new Error('RemoteAgent: \'capabilities.browserName\' is ' +
+        'required but was not specified.');
+    }
+    if (typeof options.capabilities.platform !== 'string') {
+      throw new Error('RemoteAgent: \'capabilities.platform\' is ' +
+        'required but was not specified.');
+    }
+    if (typeof options.capabilities.version !== 'string') {
+      throw new Error('RemoteAgent: \'capabilities.version\' is ' +
+        'required but was not specified.');
+    }
+    this.capabilities = Object.assign({}, options.capabilities);
+
+    if (typeof options.hostPath === 'string') {
+      throw new Error('RemoteAgent: \'hostPath\' specified.');
+    }
+  }
+
+  _createDriver() {
```

Comment:
Usually it signifies that a method is not to be used by a library's consumers. That seems to be the case here, too.

In any event, it is not a name that I chose; I am overriding a method used by [the parent class](https://github.com/bterlson/eshost/blob/ad26299b4b797f70eed03d3d3b69b5e156359e1c/lib/WebdriverAgent.js).